### ISSUE TITLE
AmazonPayClient.java not working with java 12 or higher

### ISSUE THREAD

**Issue by zhuravskiy** [amzn/amazon-pay-api-sdk-java#4]
Hello!
I am getting on java 13
```
java.lang.NoSuchFieldException: modifiers
java.lang.IllegalStateException: java.lang.NoSuchFieldException: modifiers
	at com.amazon.pay.api.AmazonPayClient.allowPatch(AmazonPayClient.java:312)
	at com.amazon.pay.api.AmazonPayClient.<init>(AmazonPayClient.java:48)
```

`allowPatch()` looks like dirty hack.

**Comment by bjguillot** [comment]
Thanks for bringing this to our attention.

You're right, the allowPatch() method is a hack to force Java into enabling the HttpURLConnection class to work with setRequestMethod("PATCH"), as it only supports GET, POST, HEAD, OPTIONS, PUT, DELETE, and TRACE.

Based on the documentation, it seems Java 12 and 13 still don't support PATCH in the HttpURLConnection class:
https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/net/HttpURLConnection.html#setRequestMethod(java.lang.String) 

We're looking into an alternate way to handle this for the newer Java versions.  Generally, our preference has been to stick with using functionality built into the core of Java when it's available and also keep compatibility with Java 8 and higher.

**Comment by zhuravskiy** [comment]
@bjguillot thank you! as a temp solution, i made downgrade to java 11 and this is works.

**Comment by bjguillot** [comment]
I came up with a fix for this.  Contact me at amazon-pay-sdk@amazon.com and I'll send you a .jar file to test.  I was able to test on OpenJDK 8, 11, 13, and 14, but I want to confirm it works in a real customer environment before publishing to GitHub.

**Comment by laplasianin** [comment]
@bjguillot Is there any updates on this?

**Comment by bjguillot** [comment]
@laplasianin, please send me an email me at the address in the comment above, and I can share a a test jar with you to see if it will work in your environment.  If it works, then I can publish the update.

**Comment by laplasianin** [comment]
laplasianin@gmail.com

**Comment by bjguillot** [comment]
Try the new jar here and let me know if it fixes your issue in Java 12+:

https://github.com/amzn/amazon-pay-api-sdk-java/releases/download/2.2.2/amazon-pay-api-2.2.2.jar

**Comment by bjguillot** [comment]
Fixed in version 2.2.2.
